### PR TITLE
Added documentation for skipping dynamic segment transition optimization

### PR DESCRIPTION
**Pull request by uberllama**
Re: #793

### CODE REVIEW

File: docs/guides/overview.md [hunk 1]
```diff
@@ -352,11 +352,36 @@ Important Note About Dynamic Segments
 If you have dynamic segments in your URL, a transition from `/users/123`
 to `/users/456` does not call `getInitialState`, `componentWillMount`, `componentWillUnmount` or `componentDidMount`. If you are using those lifecycle hooks to fetch
 data and set state, you will also need to implement
-`componentWillReceiveProps` on your handler, just like any other
+`componentWillReceiveProps` on your handler, and its child components, just like any other
```
Comment (gaearon) [inline docs/guides/overview.md]: I'd correct that to be “and its stateful children”. Stateless children don't need to implement anything.
Also nitpicking: probably don't need a comma before “and”.

File: docs/guides/overview.md [hunk 2]
```diff
@@ -352,11 +352,36 @@ Important Note About Dynamic Segments
...
 component whose props are changing. This way you can leverage the
 performance of the React DOM diff algorithm. Look at the `Contact`
 handler [in the `master-detail` example](https://github.com/rackt/react-router/blob/master/examples/master-detail/app.js).
 
+If you would rather mount components normally when transitioning between dynamic segments, you can assign a unique key to your route handler component to bypass this optimization:
```
Comment (gaearon) [inline docs/guides/overview.md]: Maybe change “mount components normally” to “force route handlers to re-mount”? This is more explicit.

File: docs/guides/overview.md [hunk 3]
```diff
@@ -352,11 +352,36 @@ Important Note About Dynamic Segments
...
+
+```js
+var App = React.createClass({
+
+  mixins: [Router.State],
+
+  getHandlerKey: function () {
+    var childDepth = 1;
```
Comment (gaearon) [inline docs/guides/overview.md]: I'd add `// assuming App is top-level route` to this line

**Comment by uberllama** [comment]
Copy updated.

**Comment by gaearon** [comment]
This is great, thank you.

**Pull request merged**